src/pytorch_lightning/trainer/trainer.py

```diff
@@ -1138,6 +1132,12 @@ def _run(
         # ----------------------------
         # TRAIN
         # ----------------------------
+        if self.state.fn in (TrainerFn.FITTING, TrainerFn.TUNING):
+            min_epochs, max_epochs = _parse_loop_limits(
+                self.min_steps, self.max_steps, self.min_epochs, self.max_epochs, self
+            )
+            self.fit_loop.min_epochs = min_epochs
+            self.fit_loop.max_epochs = max_epochs
```

Comment:
why aren't we doing it earlier? At this point we've already run model hooks that could have accessed the trainer.
Note, a few lines above we  `verify_loop_configurations(self)`, so I think this should at least move up there

Reply:
done!